ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -26,40 +29,97 @@
 import java.nio.file.Path;
 import java.security.GeneralSecurityException;
 import java.security.Key;
+import java.security.KeyFactory;
 import java.security.KeyStore;
+import java.security.PrivateKey;
 import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
 import java.security.cert.CertificateFactory;
+import java.security.spec.RSAPrivateKeySpec;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 final class KeyStores {
 
+    /**
+     * Pattern that matches a single RSA key in a PEM file. Has a capture group that captures
+     * the content of the key (everything that occurs between the header and footer).
+     */
+    private static final Pattern KEY_PATTERN = Pattern.compile(
+            "-----BEGIN RSA PRIVATE KEY-----(.+)?-----END RSA PRIVATE KEY-----", Pattern.DOTALL);
```

Comment:
keys look like this, right?
````
-----BEGIN RSA PRIVATE KEY-----
foo
-----END RSA PRIVATE KEY-----
````
then your group match will be `\nfoo\n` rather than `foo`, no?